sdks/java/core/src/main/java/org/apache/beam/sdk/options/ExperimentalOptions.java

```diff
@@ -35,6 +35,10 @@ public interface ExperimentalOptions extends PipelineOptions {
 
   String STATE_SAMPLING_PERIOD_MILLIS = "state_sampling_period_millis";
 
+  String ENABLE_CUSTOM_PUBSUB_SOURCE = "enable_custom_pubsub_source";
```

Comment:
sdk core code path does not hold information about gcp io component (pubsub experiment). Consider put these literals inside PubsubIO?

Reply:
Also, these need to be in sync with https://github.com/apache/beam/blob/618d7a8f2c6520bfae1ca84dced7fd39d7af45f4/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L552 and https://github.com/apache/beam/blob/618d7a8f2c6520bfae1ca84dced7fd39d7af45f4/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L558. If a constant is created, also refer to them there

Reply:
I removed the constants, just using literals now.

Reply:
Thanks, however, I meant because now we used this experiment in both places, it is better to define the constant in PubsubIO. google-cloud-dataflow-java depends on sdks-io-google-cloud-platform so it is fine refer to PubsubIO's constant inside DataflowRunner